quartz/components/scripts/search.inline.ts

```diff
@@ -405,11 +406,32 @@ document.addEventListener("nav", async (e: CustomEventMap["nav"]) => {
 
     let searchResults: FlexSearch.SimpleDocumentSearchResultSetUnit[]
     if (searchType === "tags") {
-      searchResults = await index.searchAsync({
-        query: currentSearchTerm.substring(1),
-        limit: numSearchResults,
-        index: ["tags"],
-      })
+      let [tag, ...queries] = currentSearchTerm.substring(1).split(" ")
+      const query = queries.join(" ").trim()
+
+      if (query !== "") {
+        // search by title and content index and then filter by tag (implemented in flexsearch)
+        searchResults = await index.searchAsync({
+          query: query,
+          // search at least 100 documents, so it is enough to filter them by tag (not tags index)
+          limit: Math.max(numSearchResults, 100),
+          index: ["title", "content"],
+          tag: tag,
+        })
+        for (let searchResult of searchResults) {
+          searchResult.result = searchResult.result.slice(0, numSearchResults)
+        }
+        // set search type to basic and remove tag from term for proper highlightning and scroll
+        searchType = "basic"
+        currentSearchTerm = query
+      } else {
+        // default search by tags index
+        searchResults = await index.searchAsync({
+          query: tag,
+          limit: numSearchResults,
+          index: ["tags"],
```

Comment:
```suggestion
          tag,
```

does this work now that we have tag as a key?

Reply:
yes, it works and we can remove the index and query fields and search only by tag: tag, but due to the implementation of the search in the flexsearch(https://github.com/nextapps-de/flexsearch/blob/4c3966709f855c6980c21035cf230b2f9150d065/src/document.js#L527-L543) , only documents with a complete tag match will be returned, and we will lose autocompletion (search by partial tag match)
because of this, I left the search by index: ["tags"] for pure tag search mode